src/coreclr/jit/gentree.cpp

```diff
@@ -15489,7 +15489,7 @@ GenTree* Compiler::gtNewTempAssign(
         if (!ok)
         {
             gtDispTree(val);
-            assert(!"Incompatible types for gtNewTempAssign");
+            noway_assert(!"Incompatible types for gtNewTempAssign");
```

Comment:
Since this is under `#ifdef DEBUG` it's not going to do anything useful.

Reply:
I see, well I will remove it then.